File: Python/그리디 알고리즘/1541.py
S = input()
S = S.split('-')
res = 0
temp = S[0].split('+')
for j in temp:
    res += int(j)

if len(S) > 1:
    for i in range(1, len(S)):
        temp = S[i].split('+')
        for j in temp:
            res -= int(j)

print(res)
    

File: BOJ/Kotlin/그리디 알고리즘/1541.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val s = readLine()
    val sSplit = s.split('-')
    var res = 0
    val temp = sSplit[0].split('+')
    for (i in temp){
        res += i.toInt()
    }

    if (sSplit.size > 1){
        for (i in 1 until sSplit.size){
            val t = sSplit[i].split('+')
            for (j in t){
                res -= j.toInt()
            }
        }
    }

    println(res)
}